src/couch_peruser/src/couch_peruser.erl

```diff
@@ -71,8 +71,10 @@
 start_link() ->
     gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
 
--spec init_state() -> #state{}.
-init_state() ->
+reset_state(#state{mem3_cluster_pid = Pid} = State) when is_pid(Pid) ->
+    unlink(Pid),
```

Comment:
Here we just unlink the process, but the process would still be alive and we'd be accumulating them and each one would still be sending stable|unstable event. The memleak is not from the links themselves but from the number of processes spawned.